assets/scripts/admin/questionEditor.js

```diff
@@ -632,76 +632,118 @@ $(document).on('ready pjax:scriptcomplete', function () {
+
+            $linkItem.addClass(classLink).attr('href', `#language_${language}`).text(languageName);
+            $liTemplate.clone().append($linkItem).appendTo($tabindex);
+
+            $bodyItem.addClass(classBody).attr('id', `language_${language}`);
+            $tabbody.append($bodyItem);
+
+            const labelSet = json.results[language];
+
+            $itemList = $listTemplate.clone();
+
+            if (labelSet.labels) {
+              isEmpty = false;
+              labelSet.labels.forEach((label) => {
+                // Label title is not concatenated directly because it may have non-encoded HTML
+                const $labelTitleDiv = $('<div class="col-md-7"></div>');
+                $labelTitleDiv.text(label.title);
+                const $listItem = $listItemTemplate.clone();
+                $listItem.append(`<div class="col-md-5 text-right" style="border-right: 4px solid #cdcdcd">${label.code}</div>`);
+                $listItem.append($labelTitleDiv);
+                $listItem.attr('data-label', JSON.stringify(label));
+                $itemList.append($listItem);
+
+                if (source === 'answeroptions' && label.code.length > 5) {
+                  hasInvalidCodes = true;
+                }
+              });
             }
-          });
 
-          $bodyItem.append(`<h4>${labelSet.label_name}</h4>`);  // jshint ignore: line
-          $itemList.appendTo($bodyItem);
-
-          if (hasInvalidCodes) {
-              $('#longcodesalert').show();
+            $bodyItem.append(`<h4>${labelSet.label_name}</h4>`);  // jshint ignore: line
+            $itemList.appendTo($bodyItem);
+          });
+          
+          if (isEmpty) {
+            showLabelSetAlert(languageJson.labelSetEmpty);
           } else {
-              $('#longcodesalert').hide();
+            if (hasInvalidCodes) {
+              showLabelSetAlert(languageJson.answeroptions.truncationWarning);
+            }
+            $('<div></div>').append($tabindex).append($tabbody).appendTo($('#labelsetpreview'));
+            $tabindex.find('li').first().find('a').trigger('click');
           }
-        });
+        }
+      },
+      error(jqXHR, textStatus, errorThrown) {
         $('#labelsetpreview').empty();
-        $('<div></div>').append($tabindex).append($tabbody).appendTo($('#labelsetpreview'));
-        $tabindex.find('li').first().find('a').trigger('click');
+        showLabelSetAlert(languageJson.labelSetFail, 'danger');
+        console.error(errorThrown);
       },
     });
   }
 
+  /**
+   * Shows an alert in the label set's modal
+   *
+   * @param {string} message The message to show
+   * @param {string=} type Alert type (eg. 'danger')
+   * @return {void}
+   */
+  function showLabelSetAlert(message /*: string */, type /*: string */) /*: void */ {
+    const alertType = type ?? 'warning';
+    var alert = $('#labelsetalert');
```

Comment:
I'd prefer if `var` was never used, personally, and always `const` or `let` instead.

Reply:
(In the LS admin view, that is. Survey taking must still support IE11, for a small while.)